src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php

```diff
@@ -0,0 +1,340 @@
+                        'route_param_name' => 'productId',
+                        'route_param_field' => 'id_product',
+                        'fragment' => 'tab-stock-tab',
+                    ])
+            );
+        }
+
+        return $columns;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getFilters()
+    {
+        $filters = (new FilterCollection())
+            ->add(
+                (new Filter('id_product', IntegerMinMaxFilterType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('id_product')
+            )
+            ->add(
+                (new Filter('name', TextType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                        'attr' => [
+                            'placeholder' => $this->trans('Search name', [], 'Admin.Catalog.Help'),
+                        ],
+                    ])
+                    ->setAssociatedColumn('name')
+            )
+            ->add(
+                (new Filter('reference', TextType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                        'attr' => [
+                            'placeholder' => $this->trans('Search reference', [], 'Admin.Catalog.Help'),
+                        ],
+                    ])
+                    ->setAssociatedColumn('reference')
+            )
+            ->add(
+                (new Filter('category', TextType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                        'attr' => [
+                            'placeholder' => $this->trans('Search category', [], 'Admin.Catalog.Help'),
+                        ],
+                    ])
+                    ->setAssociatedColumn('category')
+            )
+            ->add(
+                (new Filter('price_tax_excluded', NumberMinMaxFilterType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('price_tax_excluded')
+            )
+            ->add(
+                (new Filter('active', YesAndNoChoiceType::class))
+                    ->setAssociatedColumn('active')
+            )
+            ->add(
+                (new Filter('actions', SearchAndResetType::class))
+                    ->setTypeOptions([
+                        'reset_route' => 'admin_common_reset_search_by_filter_id',
+                        'reset_route_params' => [
+                            'filterId' => self::GRID_ID,
+                        ],
+                        'redirect_route' => 'admin_products_v2_index',
+                    ])
+                    ->setAssociatedColumn('actions')
+            );
+
+        if ($this->configuration->get('PS_STOCK_MANAGEMENT')) {
```

Comment:
do you really need to inject whole configuration?
Maybe its enough to inject bool $stockManagementOn or so?

Reply:
is there any real value in not injecting the whole configuration? I can imagine that we would need some other configuration options in the future

Reply:
Hmm while it would make code a bit cleaner. Like @kpodemski noted, what if we need other configuration options in the future? 

Then you would need to change yml, which I believe would be BC break.

Reply:
That's not the first time we have this question rising ^^
What I wonder sometime is that if we inject only the value it might be cached in symfony container's cache So if we change the configuration we need to clear the cache at each modification

By injecting the service itself which is directly connected to the DB we should keep a more dynamic behaviour without clearing the cache 🤷‍♂️